src/coreclr/runtime/portable/AllocFast.cpp

```diff
@@ -146,20 +147,105 @@ EXTERN_C FCDECL2(Object*, RhpNewPtrArrayFast, MethodTable* pMT, INT_PTR size)
 
 EXTERN_C FCDECL1(Object*, RhpNewFast, MethodTable* pMT)
 {
-    PORTABILITY_ASSERT("RhpNewFast is not yet implemented");
-    return nullptr;
+    FCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
+
+    Thread* thread = GetThread();
+    ee_alloc_context* cxt = thread->GetEEAllocContext();
+
+    size_t sizeInBytes = (size_t)pMT->GetBaseSize();
+
+    uint8_t* alloc_ptr = cxt->getAllocPtr();
+    _ASSERTE(alloc_ptr <= cxt->getAllocLimit());
+    if ((size_t)(cxt->getAllocLimit() - alloc_ptr) >= sizeInBytes)
+    {
+        cxt->setAllocPtr(alloc_ptr + sizeInBytes);
+        PtrArray* pObject = (PtrArray*)alloc_ptr;
+        pObject->SetMethodTable(pMT);
+        return pObject;
+    }
+
+    return AllocateObject(pMT, 0, 0);
 }
 
 EXTERN_C FCDECL1(Object*, RhpNewFastAlign8, MethodTable* pMT)
 {
-    PORTABILITY_ASSERT("RhpNewFastAlign8 is not yet implemented");
-    return nullptr;
+    FCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
+
+    Thread* thread = GetThread();
+    ee_alloc_context* cxt = thread->GetEEAllocContext();
+
+    size_t sizeInBytes = (size_t)pMT->GetBaseSize();
+
+    uint8_t* alloc_ptr = cxt->getAllocPtr();
+    bool requiresPadding = !IS_ALIGNED(alloc_ptr, sizeof(int64_t));
+    size_t paddedSize = sizeInBytes;
+    if (requiresPadding)
+    {
+        // We are assuming that allocation of minimal object flips the alignment
+        paddedSize += MIN_OBJECT_SIZE;
+        _ASSERTE(IS_ALIGNED(alloc_ptr + MIN_OBJECT_SIZE, sizeof(int64_t)));
+    }
+
+    _ASSERTE(alloc_ptr <= cxt->getAllocLimit());
+    if ((size_t)(cxt->getAllocLimit() - alloc_ptr) >= paddedSize)
+    {
+        cxt->setAllocPtr(alloc_ptr + paddedSize);
+        if (requiresPadding)
+        {
+            Object* dummy = (Object*)alloc_ptr;
+            dummy->SetMethodTable(g_pFreeObjectMethodTable);
+            alloc_ptr += MIN_OBJECT_SIZE;
+        }
+        PtrArray* pObject = (PtrArray*)alloc_ptr;
+        pObject->SetMethodTable(pMT);
+        return pObject;
+    }
+
+    return AllocateObject(pMT, GC_ALLOC_ALIGN8, 0);
 }
 
 EXTERN_C FCDECL1(Object*, RhpNewFastMisalign, MethodTable* pMT)
 {
-    PORTABILITY_ASSERT("RhpNewFastMisalign is not yet implemented");
-    return nullptr;
+    FCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
+
+    Thread* thread = GetThread();
+    ee_alloc_context* cxt = thread->GetEEAllocContext();
+
+    size_t sizeInBytes = (size_t)pMT->GetBaseSize();
+
+    uint8_t* alloc_ptr = cxt->getAllocPtr();
+    bool requiresPadding = IS_ALIGNED(alloc_ptr, sizeof(int64_t));
+    size_t paddedSize = sizeInBytes;
+    if (requiresPadding)
+    {
+        // We are assuming that allocation of minimal object flips the alignment
+        paddedSize += MIN_OBJECT_SIZE;
+        _ASSERTE(IS_ALIGNED(MIN_OBJECT_SIZE + sizeof(int32_t), sizeof(int64_t)));
```

Comment:
```suggestion
        paddedSize += MIN_OBJECT_SIZE;
        _ASSERTE(IS_ALIGNED(MIN_OBJECT_SIZE + sizeof(int32_t), sizeof(int64_t)));
```
```suggestion
        _ASSERTE(IS_ALIGNED(MIN_OBJECT_SIZE + sizeof(int32_t), sizeof(int64_t)));
        paddedSize += MIN_OBJECT_SIZE;
```
Nit: So that the assert is next to comment that talks about it